middleware.go

```diff
@@ -0,0 +1,100 @@
+package jwt
+
+import (
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-errors/errors"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		tok, ok := c.GetQuery("Authorization")
```

Comment:
Extract the `Authorization` token from HTTP headers instead of query parameters to enhance security and prevent token exposure in URLs. Use `c.GetHeader("Authorization")` instead of `c.GetQuery("Authorization")`. [important]